asv_bench/benchmarks/groupby.py

```diff
@@ -398,12 +360,18 @@ def setup(self, dtype, method):
                                   np.random.random(ngroups) * 10.0])
         elif dtype == 'object':
             key = ['foo'] * size
+        elif dtype == 'datetime':
+            key = date_range('1/1/2011', periods=size, freq='s')
 
         df = DataFrame({'values': values, 'key': key})
-        self.df_groupby_method = getattr(df.groupby('key')['values'], method)
+        self.as_group_method =  getattr(df.groupby('key')['values'], method)
+        self.as_field_method =  getattr(df.groupby('values')['key'], method)
+
+    def time_dtype_as_group(self, dtype, method):
+        self.as_group_method()
 
-    def time_method(self, dtype, method):
-        self.df_groupby_method()
+    def time_dtype_as_field(self, dtype, method):
```

Comment:
Before this was only testing the dtype as part of the grouping, but it makes sense to test it as an aggregated / transformed field as well